pkg/clustermanager/cluster_creator.go

```diff
@@ -0,0 +1,54 @@
+package clustermanager
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+type ClusterCreator struct {
+	Applier Applier
+	KM      KubeconfigManager
+}
+
+// CreateSync creates a workload cluster using the EKS-A controller and returns the types.Cluster object for that cluster.
+func (cc ClusterCreator) CreateSync(ctx context.Context, spec *cluster.Spec, managementCluster *types.Cluster, provider providers.Provider) (*types.Cluster, error) {
+	err := cc.Applier.Run(ctx, spec, *managementCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return cc.getWorkloadCluster(ctx, spec.Cluster.Name, managementCluster, provider)
+}
+
+func (cc ClusterCreator) getWorkloadCluster(ctx context.Context, clusterName string, management *types.Cluster, provider providers.Provider) (*types.Cluster, error) {
+	workloadCluster := &types.Cluster{
+		Name:               clusterName,
+		ExistingManagement: management.ExistingManagement,
+	}
+
+	fh, path, err := cc.KM.FS.Create(
```

Comment:
why is FS defined in KM if KM doesn't depend on it?
It seems to me that ClusterCreator is the one who is concerned about writing to disk. KM is only concerned about getting the kubeconfig and passing it to a writer.